### ISSUE TITLE
Lower boundary of `for` loop variable is not detected correctly

### ISSUE THREAD

**Issue by derrabus** [vimeo/psalm#10213]
#### Reproducer

```php
$values = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9];

for ($i = 100; $i < 110; $i++) {
    echo ((string) $values[$i - 100]) . "\n";
}
```

https://psalm.dev/r/97ca6dae65

#### Psalm's Output

ERROR: [InvalidArrayOffset](https://psalm.dev/115) - 6:20 - Cannot access value on variable $values using a int<-99, 9> offset, expecting int<0, 9>

#### Expected Output

No error

#### Notes

In the example above, Psalm correctly detects that `$i - 100` will never be greater than 9. However, it believes that the lower boundary of that operation is `-99`. Because the initial value of `$i` is 100 and because `$i` is only incremented afterwards, a value below 0 is impossible for `$i - 100`.

**Comment by weirdan** [comment]
Fixed in #10263
